api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/widgets_controller.rb

```diff
@@ -17,44 +17,45 @@ module MnoEnterprise::Concerns::Controllers::Jpi::V1::Impac::WidgetsController
   #  -> GET /api/mnoe/v1/organizations/:id/widgets
   def index
     render_not_found('organization') unless parent_organization
-    @widgets = parent_organization.widgets
+    @widgets = Mno::Widget.find(organization_id: parent_organization.id)
   end
 
   # POST /mnoe/jpi/v1/impac/dashboards/:id/widgets
   #  -> POST /api/mnoe/v1/dashboards/:id/widgets
   def create
-    if widgets
-      if @widget = widgets.create(widget_create_params)
-        MnoEnterprise::EventLogger.info('widget_create', current_user.id, 'Widget Creation', @widget)
-        @nocontent = true # no data fetch from Connec!
-        render 'show'
-      else
-        render_bad_request('create widget', @widget.errors)
-      end
+    @widget = Mno::Widget.create(widget_create_params)
+    if @widget.errors.empty?
+      MnoEnterprise::EventLogger.info('widget_create', current_user.id, 'Widget Creation', @widget)
+      @nocontent = true # no data fetch from Connec!
+      render 'show'
     else
-      render_not_found('widget')
+      render_bad_request('create widget', @widget.errors)
     end
   end
 
   # PUT /mnoe/jpi/v1/impac/widgets/:id
   #   -> PUT /api/mnoe/v1/widgets/:id
   def update
-    if widget.update(widget_update_params)
+    return render_not_found('widget') unless widget
+    widget.update(widget_update_params)
+    if widget.errors.empty
       @nocontent = !params['metadata']
       render 'show'
     else
-      render_bad_request('update widget', @widget.errors)
+      render_bad_request('update widget', widget.errors)
     end
   end
 
   # DELETE /mnoe/jpi/v1/impac/widgets/:id
   #   -> DELETE /api/mnoe/v1/widgets/:id
   def destroy
-    if widget.destroy
-      MnoEnterprise::EventLogger.info('widget_delete', current_user.id, 'Widget Deletion', widget)
+    return render_not_found('widget') unless widget
+    MnoEnterprise::EventLogger.info('widget_delete', current_user.id, 'Widget Deletion', widget)
+    widget.destroy
+    if widget.errors.empty?
```

Comment:
You got it right. widget.id is empty after you call destroy. Quite annoying actually.